sdk/python/kfp/components/executor.py

```diff
@@ -25,31 +25,43 @@ class Executor():
     """Executor executes v2-based Python function components."""
 
     def __init__(self, executor_input: Dict, function_to_execute: Callable):
-        if hasattr(function_to_execute, 'python_func'):
-            self._func = function_to_execute.python_func
-        else:
-            self._func = function_to_execute
+        self._func = function_to_execute
 
         self._input = executor_input
-        self._input_artifacts: Dict[str, artifact_types.Artifact] = {}
+        self._input_artifacts: Dict[str,
+                                    Union[artifact_types.Artifact,
+                                          List[artifact_types.Artifact]]] = {}
         self._output_artifacts: Dict[str, artifact_types.Artifact] = {}
 
         for name, artifacts in self._input.get('inputs',
                                                {}).get('artifacts', {}).items():
-            artifacts_list = artifacts.get('artifacts')
-            if artifacts_list:
-                self._input_artifacts[name] = self.make_artifact(
-                    artifacts_list[0],
-                    name,
-                    self._func,
-                )
+            list_of_artifact_proto_structs = artifacts.get('artifacts')
+            if list_of_artifact_proto_structs:
+                annotation = self._func.__annotations__[name]
+                # InputPath has no attribute __origin__ and also should be handled as a single artifact
+                if not isinstance(annotation, type_annotations.InputPath
+                                 ) and type_annotations.is_list_of_artifacts(
+                                     annotation.__origin__):
```

Comment:
Per defense programming principle, if we expect `Input` annotation here, then we should explicitly check for it, instead of checking `not Input`.

Reply:
Thanks for catching that. Updated.